website/addons/box/model.py

```diff
@@ -0,0 +1,328 @@
+# -*- coding: utf-8 -*-
+import os
+import base64
+import hashlib
+import logging
+import urllib
+from datetime import datetime
+
+import furl
+
+from modularodm import fields, Q
+from modularodm.exceptions import ModularOdmException
+
+from framework.auth import Auth
+from website.addons.base import exceptions
+from website.addons.base import AddonUserSettingsBase, AddonNodeSettingsBase, GuidFile
+
+from website.addons.box.client import get_node_addon_client
+from website.addons.box.utils import clean_path, BoxNodeLogger, refresh_creds_if_necessary
+from website.addons.box import settings
+
+from box import CredentialsV2
+
+logger = logging.getLogger(__name__)
+
+
+class BoxFile(GuidFile):
+    """A Box file model with a GUID. Created lazily upon viewing a
+    file's detail page.
+    """
+
+    #: Full path to the file, e.g. 'My Pictures/foo.png'
+    path = fields.StringField(required=True, index=True)
+
+    @property
+    def file_name(self):
+        if self.revision:
+            return '{0}_{1}_{2}.html'.format(self._id, self.revision, base64.b64encode(self.folder))
+        return '{0}_{1}_{2}.html'.format(self._id, self.unique_identifier, base64.b64encode(self.folder))
+
+    @property
+    def waterbutler_path(self):
+        path = self.path
+        if self.folder == '/':
+            return path
+        return path.replace(self.folder, '', 1)
+
+    @property
+    def folder(self):
+        return self.node.get_addon('box').folder
+
+    @property
+    def provider(self):
+        return 'box'
+
+    @property
+    def version_identifier(self):
+        return 'revision'
+
+    @property
+    def unique_identifier(self):
```

Comment:
`etag` is only supported relating to versioning if you want to delete a specific old version of a file, and even then it's optional; to prevent race conditions. Is this desired functionality?